### ISSUE TITLE
DockerStopContainer throws NotModifiedException

### ISSUE THREAD

**Issue by sbutler2901** [bmuschko/gradle-docker-plugin#586]
### Expected Behavior

It would be nice of the plugin could provide catch-able exceptions / mappings to the exceptions thrown by the underlying docker-java api client. 
 
### Current Behavior

DockerStopContainer throws the [NotModifiedException](https://github.com/docker-java/docker-java/blob/master/src/main/java/com/github/dockerjava/api/exception/NotModifiedException.java) when ran while the container is not currently running. The two successful methods I have discovered are:
1. To retrieve the exception's class and use string comparison: `if (exception.getClass().toString() == "class com.github.dockerjava.api.exception.NotModifiedException")`
2. To import the docker-java api as a buildscript dependency 
- `buildscript { .... dependencies { classpath "com.github.docker-java:docker-java:3.0.14" }}`
- `import com.github.dockerjava.api.exception.NotModifiedException`
- `if (exception.getClass() == NotModifiedException)`

### Context

This task type is used as a dependency for tasks of type DockerRemoveContainer to ensure the container is stopped before attempting to remove it.

### Steps to Reproduce (for bugs)

```
task stopDBContainer(type: DockerStopContainer) {
  description 'Stops the running database container'

  targetContainerId { createDBContainer.getContainerName() }
  onError { exception ->
    //if (exception.getClass().toString() == "class com.github.dockerjava.api.exception.NotModifiedException") {
    if (exception.getClass() == NotModifiedException) {
      println 'Container not running'
    } else if (exception.message != null && exception.message.contains('No such container')) {
      println 'Container not running'
    } else {
      throw exception
    }
  }
}
```

### Your Environment

```

------------------------------------------------------------
Gradle 4.7
------------------------------------------------------------

Build time:   2018-04-18 09:09:12 UTC
Revision:     b9a962bf70638332300e7f810689cb2febbd4a6c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          1.8.0_66 (Oracle Corporation 25.66-b17)
OS:           Mac OS X 10.13.4 x86_64
```

**Comment by orzeh** [comment]
@sbutler2901 below snippet works fine for me:

````groovy
task stopDBContainer(type: DockerStopContainer) {
  description 'Stops the running database container'
  dependsOn createMyAppContainer
  targetContainerId { createMyAppContainer.getContainerId() }
  onError { exception ->
    if (exception instanceof com.github.dockerjava.api.exception.NotModifiedException) {
      println 'Container not running'
    } else if (exception.message != null && exception.message.contains('No such container')) {
      println 'Container not running'
    } else {
      throw exception
    }
  }
}
````

**Comment by cdancy** [comment]
Yeah I do something similar to both of you but a bit simpler IMO:
```
onError { exception ->
    if (exception.class.simpleName.matches('.+(NotModifiedException|NotFoundException)$')) {
      println 'Container not running'
    } else {
      throw exception
    }
  }
```
I've found that in different situations either NotModifiedException or NotFoundException can be thrown and so we account for both. @sbutler2901 you're using things in exactly the way we designed and really there is not much we can do about the name of the exception (which I agree is not great) as the underlying `docker-java` project is not something we own..

**Comment by sbutler2901** [comment]
@orzeh how are you importing the class in your build script, if I removed the build script dependency gradle throws a compilation exception stating that it is `unable to resolve class com.github.dockerjava.api.exception.NotModifiedException`. I am using the same import statement that I provided above. 

@cdancy i like that solution since it handles both scenarios, I was just hesitant to use string comparisons to guarantee the exception's class. At least now I know I am using things correctly, and others who have this issue will be able to find our solutions quickly.
